clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+
+    ## Normalize FreeCodeCamp
+    fccWords <- c("fcc", "freecodecamp", "free code camp")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = fccWords,
+                                 replaceWith = "Free Code Camp")
+
+    ## Normalize variations of "None"
+    nones <- c("non", "none", "haven't", "havent", "not", "nothing",
+               "didn't", "n/a", "\bna\b", "never", "nil", "nope")
+    nonesIdx <- cleanPart1 %>% select(CodeEventOther) %>%
+        mutate_each(
+            funs(grepl(paste(nones, collapse = "|"),
+                       .,
+                       ignore.case = TRUE)
+                 )
+            ) %>% unlist(use.names = FALSE)
+    nonesData <- cleanPart1 %>% filter(nonesIdx) %>%
+        mutate(CodeEventOther = NA) %>%
+        mutate(CodeEventNone = "1")
+    cleanPart1 <- cleanPart1 %>% filter(!nonesIdx) %>% bind_rows(nonesData)
+
+    ## Normalize "bootcamps"
+    bootcamps <- c("^bootcamp")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = bootcamps,
+                                 replaceWith = "Bootcamp")
+
+    ## Normalize "Rails Girls"
+    railsGirls <- c("^rails? ?girls$")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = railsGirls,
+                                 replaceWith = "Rails Girls")
+
+    ## Normalize "Game Jams" to "Game Jam(s)"
+    gameJams <- c("game.*?jams?")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "CodeEventOther",
+                                 searchTerms = gameJams,
+                                 replaceWith = "Game Jam(s)")
+
+
+    # Clean Podcasts Other
+
+    ## Convert Podcasts to binary/boolean
+    podcasts <- cleanPart1 %>%
+        select(starts_with("Podcast"), -PodcastOther) %>%
+        mutate_each(funs(ifelse(!is.na(.), "1", NA)))
+    cleanPart1 <- cleanPart1 %>%
+        select(-starts_with("Podcast"), PodcastOther) %>%
+        bind_cols(podcasts)
+
+    ## Normalize variations of "None" in PodcastOther
+    nonePod <- c("non", "none", "haven't", "havent", "not a", "nothing",
+               "didn't", "n/a", "\bna\b", "never", "nil", "nope", "not tried")
+    nonesPodIdx <- cleanPart1 %>% select(PodcastOther) %>%
+        mutate_each(
+            funs(grepl(paste(nonePod, collapse = "|"),
+                       .,
+                       ignore.case = TRUE)
+            )
+        ) %>% unlist(use.names = FALSE)
+    nonesPodData <- cleanPart1 %>% filter(nonesPodIdx) %>%
+        mutate(PodcastOther = NA) %>%
+        mutate(PodcastNone = "1")
+    cleanPart1 <- cleanPart1 %>% filter(!nonesPodIdx) %>%
+        bind_rows(nonesPodData)
+
```

Comment:
There are about 60-80 mentions for each of this:

- RubyRogues <- c("rubyRogues", "ruby rogues","ruby rogue")
- DeveloperTea <- c("developer tea","developerTea")
- ShopTalk <- c("shoptalk","shop talk","shoptalk show","shop talk show")

Sometimes 2-3 podcasts with a comma

Would be good to add them, too.